doc/ensemble.rst

```diff
@@ -0,0 +1,739 @@
+The :ref:`Ensemble API<ensemble_api>` offers key Ensemble features,
+including helper functions to:
+
+- :ref:`Ensemble.add_ml_model()<load_ml_model>`: load a TF, TF-lite, PT, or ONNX model into the database at runtime.
+- :ref:`Ensemble.add_function()<load_tf_function>`: load TorchScript functions to launch with each entity belonging to the ensemble.
+- :ref:`Ensemble.add_script()<load_script>`: load TorchScripts to launch with each entity at runtime.
+- :ref:`Ensemble.enable_key_prefixing()<prefix_ensemble>`: prevent key overlapping and enable ensemble members to run the same code.
+- :ref:`Ensemble.attach_generator_files()<attach_file>`: attach configuration files to assign ensemble member parameters or load parameters to file.
+
+An Ensemble is created through the ``Experiment.create_ensemble()`` function.
+There are **three** Ensemble configuration that users can adjust to customize
+Ensemble behavior:
+
+1. :ref:`Manually<append_init>`: Create each Model individually and subsequently append the simulation to the ensemble.
+2. :ref:`Replica creation<replicas_init>`: Generate a specified number of copies or instances of a simulation.
+3. :ref:`Parameter expansion<param_expansion_init>`: Explore a range of simulation variable assignments by
+   specifying values of input parameters and selecting an expansion strategy to rearrange
+   across multiple Models within the ensemble.
+
+==============
+Initialization
+==============
+--------
+Overview
+--------
+The creation of an Ensemble involves the use of various initializer arguments,
+each serving a specific role in defining the characteristics and behavior of the
+ensemble. These arguments facilitate three distinct methods of ensemble
+creation: **manual model appending**, **parameter expansion**, and the utilization of **replicas**.
+
+The key initializer arguments are:
+
+-  `name` (Type: str): Specify the name of the ensemble, aiding in its unique identification.
+-  `params` (Type: dict[str, Any]): Provides a dictionary of parameters:values for expanding into the ``Model`` members within the ensemble. Enables parameter expansion for diverse scenario exploration.
+-  `params_as_args` (Type: list[str]): Specify which parameters from the `params` dictionary should be treated as command line arguments when executing the Models.
+-  `batch_settings` (Type: BatchSettings, optional): Describes settings for batch workload treatment.
+-  `run_settings` (Type: RunSettings, optional): Describes execution settings for individual Model members.
+-  `replicas` (Type: int, optional): Declare the number of ``Model`` clones within the ensemble, crucial for the creation of simulation replicas.
+-  `perm_strategy` (Type: str): Specifies a strategy for parameter expansion into ``Model`` instances, influencing the method of ensemble creation and number of ensemble members.
+    The options are `"all_perm"`, `"step"`, and `"random"`.
+
+-------------------
```

Comment:
Can we adjust the order of the bullet list of behaviors in the overview to be in the same order as the examples?